autoPyTorch/optimizer/smbo.py

```diff
@@ -244,10 +244,12 @@ def __init__(self,
                                               port=self.logger_port)
         self.logger.info("initialised {}".format(self.__class__.__name__))
 
-        self.initial_configurations: Optional[List[Configuration]] = None
         if portfolio_selection is not None:
-            self.initial_configurations = read_return_initial_configurations(config_space=config_space,
-                                                                             portfolio_selection=portfolio_selection)
+            initial_configurations = read_return_initial_configurations(config_space=config_space,
```

Comment:
This way makes more sense but it means there is no longer and attribute `self.initial_configurations` if `portfolio_selection` is `None`. While unlikely to matter, this means code like `if self.initial_configurations:` happening elsewhere will now break as it doesn't exist. I'd keep the default thing that it is `None` and given a value `if portfolio_selection is not None`.

Reply:
yeah, thanks for pointing it out. I'll fix this.